### PR TITLE
Feature/tagging requests

### PR DESCRIPTION
**Pull request by Cr0n1c**
Looking for an easy way to tag SDKs so that we can troubleshoot issues easier. Recommended method is to update the user-agent header with our sdk version and the platform version of the user. No sensitive data (e.g. PII) comes from this.

### CODE REVIEW

**Comment by ryfortune** [comment]
Why does the poetry.lock have so many deletions? The rest of the PR looks good

File: src/godel/api.py [hunk 1]
```diff
@@ -44,7 +48,8 @@ def __init__(
     ):
         self.url = url
         self.jwt_token = jwt_token
-        self.headers = {"Authorization": f"Bearer {jwt_token}"} if jwt_token else {}
+        self.headers = {"User-Agent": f"golden sdk v-{get_godel_version()}_{platform().lower()}", 
+                        "Authorizaiton": f"Bearer {jwt_token}"}
```
Comment (constantinmusca) [inline src/godel/api.py:52]: Authorization is misspelled

Comment (Cr0n1c) [reply src/godel/api.py:52]: sigh. South Georgian education :/

**Comment by Cr0n1c** [comment]
Found a second spot where the header could be overwritten. Since we define the headers in the __init__, I shifted it to just update that specific field and keep any other headers that it may have had. This prevents our user-agent from getting overwritten.

**Review by ryfortune — APPROVED** [review]
👍

File: pyproject.toml [hunk 2]
```diff
@@ -20,6 +20,7 @@ mkdocs = {version="^1.3.0", optional = true}
 mkdocs-material = {version="^8.2.9", optional = true}
 mkdocstrings = {version="^0.19.0", optional = true, extras=["python"]}
 mkdocs-jupyter = {version="^0.20.1", optional = true}
+toml = "^0.10.2"
```
Comment (aychang95) [inline pyproject.toml:24]: @Cr0n1c dependencies are rigid due to the data tools extra.
It's required to hit our priority to enable usage of godel in google colab and other workspaces (they use py3.7 bleh).
I'll spend some time to figure out if i can ease some dependencies

File: src/godel/api.py [hunk 3]
```diff
@@ -44,7 +48,13 @@ def __init__(
...
         self.jwt_token = jwt_token
-        self.headers = {"Authorization": f"Bearer {jwt_token}"} if jwt_token else {}
+        self.headers = {}
+        self.headers[
+            "User-Agent"
+        ] = f"golden sdk v-{get_godel_version()}_{platform().lower()}"
+        self.headers.update(
+            {"Authorization": f"Bearer {jwt_token}"} if jwt_token else {}
+        )
         self.endpoint = HTTPEndpoint(self.url, self.headers)
```
Comment (aychang95) [inline src/godel/api.py:58]: @Cr0n1c
Formatting and requiring explicit inclusion of auth header.
Having anything besides a correct token, null, or no authorization header at all outputs a vague graphql error message of "Authorization header is not of the correct bearer scheme format".
This prevents you from even successfully sending queries that don't even require auth like search, predicates, etc.

File: src/godel/api.py [hunk 4]
```diff
@@ -532,3 +544,20 @@ def add_triple_to_entity(
...
+    """
+    script_path = os.path.basename(os.path.dirname(__file__))
+
+    try:
+        pyproject_toml = toml.load(
+            os.path.join(script_path, os.pardir, os.pardir, "pyproject.toml")
+        )
+        return pyproject_toml["tool"]["poetry"]["version"]
+    except (FileNotFoundError, KeyError, toml.decoder.TomlDecodeError):
+        return "unknown"
```
Comment (aychang95) [inline src/godel/api.py:563]: cool

**Review by aychang95 — APPROVED** [review]

File: Dockerfile [hunk 5]
```diff
@@ -34,6 +34,7 @@ RUN apt-get update && \
     apt-get install -y --no-install-recommends \
     software-properties-common \
     git \
+    curl \
     git-lfs \
     ssh \
     python3.9 \
```
Comment (aychang95) [inline Dockerfile:40]: some weird changes on nvidia's hosted images...

**Review by aychang95 — APPROVED** [review]

**Pull request merged**